test/test_eos_vinet.cpp

```diff
@@ -0,0 +1,479 @@
+      constexpr std::array<Real, num> gruneisen_true{
+          B0 * A0 / Cv0 / 8.0, B0 * A0 / Cv0 / 8.5, B0 * A0 / Cv0 / 9.0,
+          B0 * A0 / Cv0 / rho0};
+
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+      // Create device views for outputs and mirror those views on the host
+      Kokkos::View<Real[num]> v_temperature("Temperature");
+      Kokkos::View<Real[num]> v_pressure("Pressure");
+      Kokkos::View<Real[num]> v_entropy("Entropy");
+      Kokkos::View<Real[num]> v_cv("Cv");
+      Kokkos::View<Real[num]> v_bulkmodulus("bmod");
+      Kokkos::View<Real[num]> v_gruneisen("Gamma");
+      auto h_temperature = Kokkos::create_mirror_view(v_temperature);
+      auto h_pressure = Kokkos::create_mirror_view(v_pressure);
+      auto h_entropy = Kokkos::create_mirror_view(v_entropy);
+      auto h_cv = Kokkos::create_mirror_view(v_cv);
+      auto h_bulkmodulus = Kokkos::create_mirror_view(v_bulkmodulus);
+      auto h_gruneisen = Kokkos::create_mirror_view(v_gruneisen);
+#else
+      // Create arrays for the outputs and then pointers to those arrays that
+      // will be passed to the functions in place of the Kokkos views
+      std::array<Real, num> h_temperature;
+      std::array<Real, num> h_pressure;
+      std::array<Real, num> h_entropy;
+      std::array<Real, num> h_cv;
+      std::array<Real, num> h_bulkmodulus;
+      std::array<Real, num> h_gruneisen;
+      // Just alias the existing pointers
+      auto v_temperature = h_temperature.data();
+      auto v_pressure = h_pressure.data();
+      auto v_entropy = h_entropy.data();
+      auto v_cv = h_cv.data();
+      auto v_bulkmodulus = h_bulkmodulus.data();
+      auto v_gruneisen = h_gruneisen.data();
+#endif // PORTABILITY_STRATEGY_KOKKOS
+
+      WHEN("A T(rho, e) lookup is performed") {
+        eos.TemperatureFromDensityInternalEnergy(v_density, v_energy, v_temperature, num);
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+        Kokkos::fence();
+        Kokkos::deep_copy(h_temperature, v_temperature);
+#endif // PORTABILITY_STRATEGY_KOKKOS
+        THEN("The returned T(rho, e) should be equal to the true value") {
+          array_compare(num, density, energy, h_temperature, temperature_true, "Density",
+                        "Energy");
+        }
+      }
+
+      WHEN("A P(rho, e) lookup is performed") {
+        eos.PressureFromDensityInternalEnergy(v_density, v_energy, v_pressure, num);
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+        Kokkos::fence();
+        Kokkos::deep_copy(h_pressure, v_pressure);
+#endif // PORTABILITY_STRATEGY_KOKKOS
+        THEN("The returned P(rho, e) should be equal to the true value") {
+          array_compare(num, density, energy, h_pressure, pressure_true, "Density",
+                        "Energy");
+        }
+      }
+
+      WHEN("A S(rho, e) lookup is performed") {
+        for (int i = 0; i < num; i++) {
+          v_entropy[i] = eos2.EntropyFromDensityInternalEnergy(v_density[i], v_energy[i]);
+        }
```

Comment:
```suggestion
        portableFor("Test S(rho,e) points", 0, num - 1, PORTABLE_LAMBDA(int i) {
          v_entropy[i] = eos2.EntropyFromDensityInternalEnergy(v_density[i], v_energy[i]);
        });
```
The Darwin CI fails in this test. Maybe this has to do with the lack of `portableFor`?

Reply:
I just updated this to make it a portable for. I'm re-running the Darwin-CI.

Sorry to manually tweak your PR @aematts. I think this is faster turnaround though. The `portableFor` lets it run on GPU instead of CPU. The GPU tests are probably failing because `eos2` and the `v_density` array live on GPU memory and accessing them from a CPU "execution space" causes errors. The `portableFor` moves it to GPU execution space.

Reply:
appears to have passed Darwin CI!